src/components/ComponentSettingsModal.jsx

```diff
@@ -0,0 +1,363 @@
+      title,
+      date,
+      allCategories,
+      permalink,
+      fileUrl,
+      errors,
+      canShowDeleteWarningModal,
+      sha,
+      redirectToNewPage,
+      newPageUrl,
+      thirdNavTitle,
+      isPost,
+    } = this.state;
+    const { settingsToggle, isNewFile, type, modalTitle, siteName } = this.props;
+
+    // Settings form has errors - disable save button
+    const hasErrors = _.some(errors, (field) => field.length > 0);
+
+    return (
+      <>
+        <div className={elementStyles.overlay}>
+          { (sha || isNewFile)
+          && (
+          <div className={elementStyles['modal-settings']}>
+            <div className={elementStyles.modalHeader}>
+              <h1>{modalTitle}</h1>
+              <button id="settings-CLOSE" type="button" onClick={settingsToggle}>
+                <i id="settingsIcon-CLOSE" className="bx bx-x" />
+              </button>
+            </div>
+            <div className={elementStyles.modalContent}>
+              <div className={elementStyles.modalFormFields}>
+                {/* Category */}
+                <p className={elementStyles.formLabel}>Category Folder Name</p>
+                <div className="d-flex">
+                  <select className="w-100" id="category" value={category} onChange={this.changeHandler} disabled={!isNewFile}>
+                    {
+                    allCategories
+                      ? allCategories.map((category) => (
+                        <option
+                          value={category}
+                          label={category}
+                        />
+                      ))
+                      : null
+                  }
+                  </select>
+                </div>
+                {/* Title */}
+                <FormField
+                  title="Title"
+                  id="title"
+                  value={title}
+                  errorMessage={errors.title}
+                  isRequired
+                  onFieldChange={this.changeHandler}
+                />
+                {/* Third Nav */}
+                { 
+                  thirdNavTitle &&
+                  <FormField
+                    title="3rd Nav Title"
+                    id="thirdNavTitle"
+                    value={thirdNavTitle}
+                    onFieldChange={this.changeHandler}
+                  />
+                }
+                {/* Permalink */}
+                <FormField
+                  title="Permalink"
+                  id="permalink"
+                  value={permalink ? permalink : ''}
+                  errorMessage={errors.permalink}
+                  isRequired
+                  onFieldChange={this.changeHandler}
+                  disabled={!isPost}
+                  fixedMessage={category ? `/${category}/${thirdNavTitle ? `${thirdNavTitle}/` : ''}` : '/'}
+                  placeholder=' '
+                />
+                { type === "resource" && 
+                <ResourceFormFields 
```

Comment:
missing `isRequired` prop

Reply:
Removed the isRequired prop requirement in ResourceFormFields, as it should be optional. Resolved in https://github.com/isomerpages/isomercms-frontend/pull/184/commits/8aaeadf1e42fcc76f360acaa10de58f3e645fb18